lbz/events/api.py

```diff
@@ -60,28 +43,28 @@ def set_bus_name(self, bus_name: str) -> None:
         self._bus_name = bus_name
 
     @property
-    def sent_events(self) -> List[BaseEvent]:
+    def sent_events(self) -> List[Event]:
         return deepcopy(self._sent_events)
 
     @property
-    def pending_events(self) -> List[BaseEvent]:
+    def pending_events(self) -> List[Event]:
         return deepcopy(self._pending_events)
 
     @property
-    def failed_events(self) -> List[BaseEvent]:
+    def failed_events(self) -> List[Event]:
         return deepcopy(self._failed_events)
 
-    def register(self, new_event: BaseEvent) -> None:
+    def register(self, new_event: Event) -> None:
         self._pending_events.append(new_event)
 
     # TODO: Stop sharing protected lists outside the class, use the above properties instead
-    def get_all_pending_events(self) -> List[BaseEvent]:
+    def get_all_pending_events(self) -> List[Event]:
```

Comment:
After your changes, we will need to adjust the rest of our code anyway, so I think this is a good moment to remove these methods 😉

Reply:
I think we agreed to do it with 0.6.0

Reply:
And as far as I checked it should not be a breaking change.